src/main/java/com/google/sps/daos/DatabaseRoomDao.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.daos;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firestore.v1.StructuredQuery.CollectionSelector;
+
+public class DatabaseRoomDao implements RoomDao {
+
+  private DatabaseReference roomsDBReference;
+  private Firestore firestoreDb;
+  private String actualRoom;
+  private String gameId;
+
+  public DatabaseRoomDao(Firestore firestoreDb) {
+    this.firestoreDb = firestoreDb;
+  }
+  
+  @Override
+  public String createNewRoom(String gameId) {
+
+    this.gameId = gameId;
+    DocumentReference newRoomRef = firestoreDb.collection("liveRooms").document();
+
+ 
+    String userId = "pamelalozano";
+
+    Map<String, Object> data = new HashMap<>();
+    data.put("creator", userId);
+    data.put("gameId", gameId);
+    data.put("isActive", false);
+
+    newRoomRef.set(data);
+
+    //Room document is now created
+    this.actualRoom = newRoomRef.getId();
+
+   //Updates game document and adds liveRoom id
+   Map<String, Object> update = new HashMap<>();
+   update.put("liveRoom", this.actualRoom);
+   ApiFuture<WriteResult> writeResult = firestoreDb.collection("games").document(this.gameId).set(update, SetOptions.merge());
+
+
+    return this.actualRoom;
+  }
+ 
+ @Override
+  public String getActualRoom() {
+      return this.actualRoom;
+  }
+
+  @Override
```

Comment:
I feel like there should be a separation between `RoomDao` and a `Room` where `RoomDao` just deals with CRUD operations on a `Room`, and `Room` is just a POJO.

`joinRoom` and `startGame` seem like they could both be represented as "updates" to a `Room` entity.

Reply:
@brettapeters so if the purpose of `joinRoom` is to just add a member to the DB's list of members for a given room so that the client can listen to it how do you think we should separate? Maybe in the Servlet creating a `Room` instance that represents the actual state of the room in the DB and update it there and then save the updated `Room` instance to the DB?

Reply:
Yeah so I think Brett means creating a Room entity and then using the Dao to modify it, like the questions example in the design doc, and once modified local pass the update of the Room entity to the db ? @brettapeters